sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -218,6 +218,30 @@ struct ImageDeleter {
   }
 };
 
+static bool is_compiler_available(const context &C,
+                                  RT::PiDeviceBinaryType Format) {
+  // SPIR-V needs to be able to compile IL
+  if (Format != PI_DEVICE_BINARY_TYPE_SPIRV)
```

Comment:
If I understand the aim of this function correctly - it should return `false` if `Format` is not supported and `true` otherwise. Right?
The unsupported cases are:
- SPIR-V format if OpenCL version is < 2.1 and there are not devices supporting `cl_khr_program_il` extension.
- What about `PI_DEVICE_BINARY_TYPE_LLVMIR_BITCODE`?
I assume `PI_DEVICE_BINARY_TYPE_NONE` is already handled and `PI_DEVICE_BINARY_TYPE_NATIVE` is always supported. Right?

I think the comment about this `if` statement should be updated to say something like "All formats except `PI_DEVICE_BINARY_TYPE_SPIRV` are supported."

Reply:
To be honest, I am not sure what to do about PI_DEVICE_BINARY_TYPE_LLVMIR_BITCODE.  That isn't a supported OpenCL IR.  I am open to suggestions.  Certainly the Intel FPGA will **not** support LLVM BC.